includes/REST_API/Media_Hotlinking_Controller.php

```diff
@@ -0,0 +1,300 @@
+
+		$file_name = basename( $path );
+		$data      = [
+			'ext'       => $ext,
+			'file_name' => $file_name,
+			'file_size' => $file_size,
+			'mime_type' => $mime_type,
+		];
+
+		set_transient( $cache_key, wp_json_encode( $data ), $cache_ttl );
+
+		$response = $this->prepare_item_for_response( $data, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+
+	/**
+	 * Prepares response asset response.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @param array           $link URL data value, default to false is not set.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object.
+	 */
+	public function prepare_item_for_response( $link, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$schema = $this->get_item_schema();
+
+		$data = [];
+
+		$error = new WP_Error();
+		foreach ( $schema['properties'] as $field => $args ) {
+			if ( rest_is_field_included( $field, $fields ) ) {
+				$check = rest_validate_value_from_schema( $link[ $field ], $args, $field );
+				if ( is_wp_error( $check ) ) {
+					$error->add( 'rest_invalid_' . $field, $check->get_error_message(), [ 'status' => 400 ] );
+				}
+
+				$data[ $field ] = rest_sanitize_value_from_schema( $link[ $field ], $args, $field );
+			}
+		}
+
+		if ( $error->get_error_codes() ) {
+			return $error;
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the link's schema, conforming to JSON Schema.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema(): array {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$allowed_mime_types = $this->get_allowed_mime_types();
+		$allowed_mime_types = array_merge( ...array_values( $allowed_mime_types ) );
+		$exts               = $this->get_file_type_exts( $allowed_mime_types );
+
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'link',
+			'type'       => 'object',
+			'properties' => [
+				'ext'       => [
+					'description' => __( 'File extension', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+					'enum'        => $exts,
+				],
+				'file_name' => [
+					'description' => __( 'File name', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'file_size' => [
+					'description' => __( 'File size', 'web-stories' ),
+					'type'        => 'integer',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'mime_type' => [
+					'description' => __( 'Mime Type', 'web-stories' ),
```

Comment:
Nit: 

```suggestion
					'description' => __( 'Mime type', 'web-stories' ),
```

Since the other descriptions are not capitalized either